### DISCUSSION TITLE
Variant groups?

### DISCUSSION THREAD

**Discussion by theoephraim** [histoire-dev/histoire#192] [Ideas]
Just starting to get set up and already thinking another level of hierarchy might be useful in some situations.

For example, lets say my button component has some props, like type (primary, secondary, error, etc), size (s, m, l, etc), style (solid, outline, transparent)

Of course having a single sort of "playground" story that uses the controls and can do anything is awesome. But I also want to show all various options grouped together somehow.

One option is to have a single story called "sizes" which has one button of each size it. That works, but probably best to keep a single button per "Variant". Problem is I'd like those size related variants grouped together somehow, but ideally not split into a new story that is only about button sizes.

Perhaps introducing a new concept of a "variant group" could be helpful. In a grid view, each group could be displayed slightly split from one another, or could even show the secondary nav like when viewing single stories.

Open to other ideas, or patterns others have found useful.

Thanks!

**Comment by reslear** [comment]
approve, must!

name like `grid-group`

before:

<img width="897" alt="Screenshot 2024-05-06 at 20 37 55" src="https://github.com/histoire-dev/histoire/assets/12596485/35ad92e6-dc55-4d90-a3eb-14fb5905a350">

after:

<img width="851" alt="Screenshot 2024-05-06 at 20 42 05" src="https://github.com/histoire-dev/histoire/assets/12596485/6b868c1f-0760-43c7-b864-96afa46b072a">

need only change to multiple grids and add headers:

<img width="742" alt="Screenshot 2024-05-06 at 20 42 58" src="https://github.com/histoire-dev/histoire/assets/12596485/ce139b2f-642f-4999-8c11-691cd7c4eb5f">


hi @Akryum maybe it's not too hard to implement?
